pkg/kubelet/dockershim/docker_container.go

```diff
@@ -162,11 +162,20 @@ func (ds *dockerService) CreateContainer(_ context.Context, r *runtimeapi.Create
 
 	hc.SecurityOpt = append(hc.SecurityOpt, securityOpts...)
 
+	cleanupInfo, err := ds.applyPlatformSpecificDockerConfig(r, &createConfig)
+	if err != nil {
+		return nil, err
+	}
+
 	createResp, err := ds.client.CreateContainer(createConfig)
 	if err != nil {
 		createResp, err = recoverFromCreationConflictIfNeeded(ds.client, createConfig, err)
 	}
 
+	if err = ds.performPlatformSpecificContainerCreationCleanup(cleanupInfo); err != nil {
```

Comment:
Assigning to `err` above is going to mask any previous errors that `err` was populated with from `ds.client. CreateContainer()` or `recoverFromCreationConflictIfNeeded`. I think `TestContainerCreationConflict` in pull-kubernetes-bazel-test is failing because of this. Can you refactor this a bit to preserve `err` from the earlier steps and return it?

Reply:
Good catch, wish I had read this before wasting 30 mins debugging it ;)